pkg/kubelet/kubeletconfig/util/codec/codec.go

```diff
@@ -26,9 +26,37 @@ import (
 	"k8s.io/apimachinery/pkg/runtime/serializer"
 	"k8s.io/kubernetes/pkg/api/legacyscheme"
 	"k8s.io/kubernetes/pkg/kubelet/apis/kubeletconfig"
+	"k8s.io/kubernetes/pkg/kubelet/apis/kubeletconfig/scheme"
+	"k8s.io/kubernetes/pkg/kubelet/apis/kubeletconfig/v1beta1"
 )
 
-// TODO(mtaufen): allow an encoder to be injected into checkpoint objects at creation time? (then we could ultimately instantiate only one encoder)
+// EncodeKubeletConfig encodes an internal KubeletConfiguration to an external JSON representation
+func EncodeKubeletConfig(internal *kubeletconfig.KubeletConfiguration) ([]byte, error) {
+	encoder, err := newKubeletConfigJSONEncoder()
+	if err != nil {
+		return nil, err
+	}
+	// encoder will convert to external version
+	data, err := runtime.Encode(encoder, internal)
+	if err != nil {
+		return nil, err
+	}
+	return data, nil
+}
+
+// newKubeletConfigJSONEncoder returns an encoder that can write a KubeletConfig to JSON
+func newKubeletConfigJSONEncoder() (runtime.Encoder, error) {
+	_, codecs, err := scheme.NewSchemeAndCodecs()
+	if err != nil {
+		return nil, err
+	}
+	mediaType := "application/json"
```

Comment:
I know this is a straight move, but did we want json or yaml? not a blocking issue, just curious... config files elsewhere have preferred yaml

Reply:
Agree we should probably update this to use yaml.